src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -15,6 +15,11 @@ public class ToolSettings
         public FilePath ToolPath { get; set; }
 
         /// <summary>
+        /// Gets or sets optional timeout for tool execution
+        /// </summary>
+        public int? ToolTimeout { get; set; }
```

Comment:
I'm OK with TimeSpanInMilliseconds as well. Not going to push this if you think it's not a good idea :smile: